apps/api/src/config/cors.ts

```diff
@@ -12,29 +12,18 @@ export const corsOptionsDelegate: Parameters<INestApplication['enableCors']>[0]
 
   const origin = (req.headers as any)?.origin || '';
 
-  if (['test', 'local'].includes(process.env.NODE_ENV) || isWidgetRoute(req.url) || isBlueprintRoute(req.url)) {
+  if (
+    ['test', 'local'].includes(process.env.NODE_ENV) ||
+    isWidgetRoute(req.url) ||
+    isBlueprintRoute(req.url) ||
+    hasPermittedDeployPreviewOrigin(origin)
```

Comment:
<!-- Notes are always non-blocking and simply highlight something the reader should take note of.  -->
🗒 **note (non-blocking):** I felt this cleaned up the code to leverage the same pattern and reuse the origin that we _know‏_ is working